clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -269,7 +269,8 @@ private void executeCommandWithRetry(
       Supplier<Boolean> remoteCommandSupplier,
       Runnable localCommandRunnable) {
     do {
-      if (isTopicPartitionHostedInMainProcess(topicName, partition)) {
+      if (isTopicPartitionHostedInMainProcess(topicName, partition) || (!isTopicPartitionIngesting(topicName, partition)
+          && (!command.equals(IngestionCommandType.START_CONSUMPTION)))) {
```

Comment:
Nitpick: this kind of multi-part boolean equation can be a bit hard to read... it may help readability to reduce it a bit. Some suggestions include:

1. The third boolean is wrapped in unnecessary parenthesis, which can be removed.
2. Adding a static import for `IngestionCommandType.START_CONSUMPTION` would allow you to write just `START_CONSUMPTION` which is sufficiently precise to understand, but shorter.
3. Enums don't need `equals` and can be compared with `==` (or in this case, `!=`, since you're negating).

End result could be something like:

```
      if (isTopicPartitionHostedInMainProcess(topicName, partition) || (!isTopicPartitionIngesting(topicName, partition)
           && command != START_CONSUMPTION)) {
```
Or (if spotless allows is), perhaps:

```
      if (isTopicPartitionHostedInMainProcess(topicName, partition) 
            || (!isTopicPartitionIngesting(topicName, partition) && command != START_CONSUMPTION)) {
```

BTW, can you think of a way for this logic change to be unit testable?

Reply:
It could be. Let me issue a bunch of wc-tests after address those comments to make sure no more regression and I will add unit test at the same time.